src/shear.c

```diff
@@ -238,26 +238,72 @@ void shear_procpair(Shear* self,
     double sin2pa = sin(2*posangle_radians);
 
 
-    // note we already checked if lens z was in our interpolation range
-    double scinv;
-    // scinv is used for weighting and gamma->DeltaSigma except for 
-    // SIGMACRIT_STYLE_SAMPLE, where it is only used for weighting
-    if (config->scstyle == SIGMACRIT_STYLE_INTERP) {
-        scinv = interplin(src->zlens, src->scinv, lens->z);
-    } else {
-        if ( (src->z - lens->z) < config->zdiff_min) {
-            goto _procpair_bail;
-        }
-        double dcl = lens->da*(1.+lens->z);
-        scinv = scinv_pre(lens->z, dcl, src->dc);
-    }
-
-    if (scinv <= 0) {
-        fprintf(stderr, "%ld %ld is a source-lens pair with non-positive inverse sigma_crit\n", lens->index, src->index);
-        fprintf(stderr, "check that your zdiff_min is set to something > 0 in the config file\n");
-        goto _procpair_bail;
-    }
-
+    // philosophy:
+    // any entry in our data vector should be a weighted mean of tangential shear,
+    // potentially normalized to be in DeltaSigma units
+    //
+    // i.e., 
+    // G=\sum_i(w_i*g_i)
+    // g=G/W where W=\sum_i(w_i) is a unitless mean shear
+    // DeltaSigma=G/S where S=\sum_i(w_i*sigma_crit^-1_i) is an estimate of DeltaSigma in physical units (Msol/Mpc^2)
+    //
+    // so we always output three columns: G, W=\sum_i(w_i) and S=\sum_i(w_i*sigma_crit_i^-1_i)
+    //
+    // we allow two different weighting style: config->weightstyle=uniform and config->weightstyle=optimal
+    // 'uniform' will use the weights as written in the input catalog (shape noise weights)
+    // 'optimal' will use these times sigma_crit_i^-1 as the weight, which maximizes the S/N
+	// 
+	// the config->shear_units parameter is obsolete because we always output both W and S
+	
+	
+	// (1) set sigma_crit^-1
+	double scinv; // the sigma_crit^-1 that goes into S
+	switch(config->scstyle) {
+		case SIGMACRIT_STYLE_INTERP:
+			scinv = interplin(src->zlens, src->scinv, lens->z);
+			break;
+		case SIGMACRIT_STYLE_POINT:
+        	if ( (src->z - lens->z) < config->zdiff_min) {
```

Comment:
some indentation problems here:  are you using tabs? If so please convert to spaces and follow my original style.